# original AVX and AVX2 tests were passed by icc compiler, the new code works well for both gcc and icc
# May 11th 2018 by B. Hajgato (Free Univeristy Brussles - VUB)
--- pytorch-0.3.1/torch/lib/TH/cmake/FindSSE.cmake.orig	2018-02-09 18:07:43.000000000 +0100
+++ pytorch-0.3.1/torch/lib/TH/cmake/FindSSE.cmake	2018-05-11 12:00:36.646562052 +0200
@@ -59,22 +59,40 @@
 
 SET(AVX_CODE "
   #include <immintrin.h>
-
   int main()
   {
-    __m256 a;
-    a = _mm256_set1_ps(0);
+    __m256 a, b, c;
+    const float src[8] = { 1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f };
+    float dst[8];
+    a = _mm256_loadu_ps( src );
+    b = _mm256_loadu_ps( src );
+    c = _mm256_add_ps( a, b );
+    _mm256_storeu_ps( dst, c );
+    for( int i = 0; i < 8; i++ ){
+      if( ( src[i] + src[i] ) != dst[i] ){
+        return -1;
+      }
+    }
     return 0;
   }
 ")
 
 SET(AVX2_CODE "
   #include <immintrin.h>
-
   int main()
   {
-    __m256i a = {0};
-    a = _mm256_abs_epi16(a);
+    __m256i a, b, c;
+    const int src[8] = { 1, 2, 3, 4, 5, 6, 7, 8 };
+    int dst[8];
+    a =  _mm256_loadu_si256( (__m256i*)src );
+    b =  _mm256_loadu_si256( (__m256i*)src );
+    c = _mm256_add_epi32( a, b );
+    _mm256_storeu_si256( (__m256i*)dst, c );
+    for( int i = 0; i < 8; i++ ){
+      if( ( src[i] + src[i] ) != dst[i] ){
+        return -1;
+      }
+    }
     return 0;
   }
 ")
